### ISSUE TITLE
Error creating an RDS instance

### ISSUE THREAD

**Issue by zleswomp** [puppetlabs/puppetlabs-aws#302]
With a resource like so:

``` puppet
  rds_instance { 'somerdsinstance':
    engine               => 'mysql',
    engine_version       => '5.6.27',
    master_username      => 'username',
    master_user_password => $password,
  }
```

I get the following error:

```
Error: Failed to apply catalog: unable to sign request without credentials set
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/request_signer.rb:100:in `require_credentials'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/request_signer.rb:90:in `sign_authenticated_requests'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/request_signer.rb:83:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/retry_errors.rb:87:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/query/handler.rb:27:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/user_agent.rb:12:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/seahorse/client/plugins/endpoint.rb:41:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/param_validator.rb:21:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/seahorse/client/plugins/raise_response_errors.rb:14:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/aws-sdk-core/plugins/response_paging.rb:26:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/seahorse/client/plugins/response_target.rb:21:in `call'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/seahorse/client/request.rb:70:in `send_request'
/opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/gems/aws-sdk-core-2.2.37/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
/opt/puppetlabs/puppet/cache/lib/puppet/provider/rds_instance/v2.rb:11:in `block in instances'
/opt/puppetlabs/puppet/cache/lib/puppet/provider/rds_instance/v2.rb:9:in `collect'
/opt/puppetlabs/puppet/cache/lib/puppet/provider/rds_instance/v2.rb:9:in `instances'
/opt/puppetlabs/puppet/cache/lib/puppet/provider/rds_instance/v2.rb:29:in `prefetch'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/transaction.rb:305:in `prefetch'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/transaction.rb:205:in `prefetch_if_necessary'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/transaction.rb:97:in `block in evaluate'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/graph/relationship_graph.rb:116:in `call'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/graph/relationship_graph.rb:116:in `traverse'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/transaction.rb:142:in `evaluate'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/resource/catalog.rb:221:in `block in apply'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/log.rb:155:in `with_destination'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/transaction/report.rb:118:in `as_logging_destination'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/resource/catalog.rb:220:in `apply'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:171:in `block in apply_catalog'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:223:in `block in benchmark'
/opt/puppetlabs/puppet/lib/ruby/2.1.0/benchmark.rb:294:in `realtime'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:222:in `benchmark'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:170:in `apply_catalog'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:315:in `run_internal'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:186:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:240:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:185:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:45:in `block (4 levels) in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent/locker.rb:21:in `lock'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:45:in `block (3 levels) in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:98:in `with_client'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:42:in `block (2 levels) in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:65:in `run_in_fork'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:41:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:179:in `call'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:179:in `controlled_run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/agent.rb:39:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/agent.rb:353:in `onetime'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/agent.rb:331:in `run_command'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:540:in `exit_on_fail'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:128:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:72:in `execute'
/opt/puppetlabs/puppet/bin/puppet:5:in `<main>'
```

It should be noted that I have the credentials set in `/etc/puppetlabs/puppet/puppetlabs_aws_credentials.ini` and am able to deploy other aws resources without issue.

**Comment by zleswomp** [comment]
I have worked around this issue by executing the following:
```
mkdir ~/.aws/
cp /etc/puppetlabs/puppet/puppetlabs_aws_credentials.ini ~/.aws/credentials
chmod 0600 ~/.aws/credentials
```
Storing the credentials in two places seems like a bad requirement.  Can someone enlighten me on the way credentials *should* be managed?  I had thought I only needed the `puppetlabs_aws_credentials.ini` file due to this line: https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet_x/puppetlabs/aws.rb#L78

**Comment by hesamaseh** [comment]
This pull request is going to fix that 

https://github.com/puppetlabs/puppetlabs-aws/pull/314

**Comment by zleswomp** [comment]
Thank you @hesamaseh

**Comment by ahenroid** [comment]
@zleswomp I believe this was fixed by #314. Can I close the issue? Thanks!

**Comment by zleswomp** [comment]
Yes, this has been resolved.
